pkg/controller/release/application.go

```diff
@@ -1,223 +0,0 @@
-	patches, transitions, err := strategyExecutor.Execute()
-	if err != nil {
-		releaseSyncedCond := apputil.NewApplicationCondition(
-			shipper.ApplicationConditionTypeReleaseSynced,
-			corev1.ConditionFalse,
-			conditions.StrategyExecutionFailed,
-			fmt.Sprintf("failed to execute application strategy: %q", err),
-		)
-		diff := apputil.SetApplicationCondition(&app.Status, *releaseSyncedCond)
-		_, err = c.clientset.ShipperV1alpha1().Applications(app.Namespace).Update(app)
-		if err != nil {
-			return shippererrors.NewKubeclientUpdateError(app, err).
-				WithShipperKind("Application")
-		}
-		if !diff.IsEmpty() {
-			c.recorder.Eventf(app, corev1.EventTypeNormal, "ApplicationConditionChanged", diff.String())
-		}
-		return err
-	}
-
-	for _, t := range transitions {
-		c.recorder.Eventf(
-			strategyExecutor.contender.release,
-			corev1.EventTypeNormal,
-			"ReleaseStateTransitioned",
-			"Release %q had its state %q transitioned to %q",
-			shippercontroller.MetaKey(strategyExecutor.contender.release),
-			t.State,
-			t.New,
-		)
-	}
-
-	if len(patches) == 0 {
-		klog.V(4).Infof("Strategy verified, nothing to patch")
-		return nil
-	}
-
-	klog.V(4).Infof("Strategy has been executed, applying patches")
-	for _, patch := range patches {
-		name, gvk, b := patch.PatchSpec()
-
-		var err error
-		switch gvk.Kind {
-		case "Release":
-			_, err = c.clientset.ShipperV1alpha1().Releases(namespace).Patch(name, types.MergePatchType, b)
-		case "InstallationTarget":
-			_, err = c.clientset.ShipperV1alpha1().InstallationTargets(namespace).Patch(name, types.MergePatchType, b)
-		case "CapacityTarget":
-			_, err = c.clientset.ShipperV1alpha1().CapacityTargets(namespace).Patch(name, types.MergePatchType, b)
-		case "TrafficTarget":
-			_, err = c.clientset.ShipperV1alpha1().TrafficTargets(namespace).Patch(name, types.MergePatchType, b)
-		default:
-			return shippererrors.NewUnrecoverableError(fmt.Errorf("error syncing Application %q (will not retry): unknown GVK resource name: %s", key, gvk.Kind))
-		}
-		if err != nil {
-			return shippererrors.NewKubeclientPatchError(namespace, name, err).WithKind(gvk)
-		}
-	}
-
-	return nil
-}
-
-func (c *Controller) buildExecutor(incumbentRelease, contenderRelease *shipper.Release) (*Executor, error) {
-	if !releaseutil.ReleaseScheduled(contenderRelease) {
-		return nil, shippererrors.NewNotWorkingOnStrategyError(shippercontroller.MetaKey(contenderRelease))
-	}
-
-	contenderReleaseInfo, err := c.buildReleaseInfo(contenderRelease)
-	if err != nil {
-		return nil, err
-	}
-
-	strategy := *contenderReleaseInfo.release.Spec.Environment.Strategy
-
-	// No incumbent, only this contender: a new application.
-	if incumbentRelease == nil {
-		return &Executor{
-			contender: contenderReleaseInfo,
-			recorder:  c.recorder,
-			strategy:  strategy,
-		}, nil
-	}
-
-	incumbentReleaseInfo, err := c.buildReleaseInfo(incumbentRelease)
-	if err != nil {
-		return nil, err
-	}
-
-	return &Executor{
```

Comment:
Unless I missed it, looks like the old `Executor` is not used anymore. Should we delete it as well?